src/core/util/xlio_list.h

```diff
@@ -75,119 +74,23 @@ template <class T, size_t offset(void)> class list_node {
 #endif
 
     list_node()
-        : obj_ptr(NULL)
     {
         this->head.next = &this->head;
         this->head.prev = &this->head;
         VLIST_DEBUG_SET_PARENT(this, NULL);
     }
 
+    T *obj_ptr() { return reinterpret_cast<T *>(reinterpret_cast<uintptr_t>(this) - offset()); }
+
     /* is_list_member - check if the node is already a member in a list. */
     bool is_list_member()
     {
         return this->head.next != &this->head || this->head.prev != &this->head;
     }
 };
 
-template <typename T, size_t offset(void)>
```

Comment:
I believe we can rewrite the iterator to use obj_ptr() method. But not critical.